jina/resources/multimodal/pods/segmenter.py

```diff
@@ -22,6 +24,7 @@ def craft(self, tags):
 class BiSegmenter(Segmenter):
     """Segmenter for multimodal example."""
 
+    @single_multi_input(num_data=2, flatten_output=False)
```

Comment:
This setting increases the learning time for developers. They not only need to learn which exec to choose, but also determine how to decorate the method.
Is it possible to maintain the previous code and behavior, and we only add these decorators when needed?

Reply:
I would love to do so, but the cost for this would mean having an extra set of Drivers, (with an explosion of Drivers).

One thing I wanted to do was to see if we can add the decorators automatically.

But yes, it is a problem from this point of view. 

What can we do about this @nan-wang ?